Enable transaction support in integration tests for cardano-http-bridge

KtorZ opened the pull request:
# Issue Number

<!-- Put here a reference to the issue this PR relates to and which requirements it tackles -->

#220 

# Overview

<!-- Detail in a few bullet points the work accomplished in this PR -->

- [ ] I have slightly improved error message in the networking layer when an unexpected error is encountered
- [ ] I have fixed the local cluster installation so that "local" would actually match "testnet" configuration but with a different genesis UTxO upon which we have some control.
- [ ] I have allowed a special transaction in the integration test framework that is made from a rich faucet and can be used to provide many wallets with funds.

# Comments

<!-- Additional comments or screenshots to attach if any -->

To be done: initialize a hundred wallets with funds by making an initial "setup" transaction to addresses we control (with known mnemonics). Later, these wallets can be restored and used in integration scenarios.

<!-- 
Don't forget to:

 ✓ Self-review your changes to make sure nothing unexpected slipped through
 ✓ Assign yourself to the PR
 ✓ Assign one or several reviewer(s)
 ✓ Once created, link this PR to its corresponding ticket
 ✓ Acknowledge any changes required to the Wiki
-->

KtorZ committed:
add _debug_ output when restoring transactions, display all restored txs metas

KtorZ committed:
Improve debugging capability for unexpected network failures by providing some context about the request

KtorZ committed:
review overall integration cluster startup delays and folders

KtorZ committed:
review local cluster configuration to 'match' testnet as closely as possible

KtorZ committed:
write helper to extract funds from a faucet address and a first dummy test to show it works

KtorZ committed:
add missing dependencies and modules to http-bridge dot cabal

piotr-iohk commented on lib/http-bridge/test/integration/Test/Integration/Faucet.hs:
```diff
@@ -0,0 +1,125 @@
+
+-- | A faucet / genesis UTxO generate from the configuration.yaml.
+-- The secret key can be generated using `cardano-keygen`, and then, using
+-- `deriveForstHDAddress` from cardano-sl to get the corresponding address key.
```
`deriveFirstHDAddress` ?

piotr-iohk commented on lib/http-bridge/test/integration/Test/Integration/Scenario/Transactions.hs:
```diff
@@ -0,0 +1,105 @@
+        -- | Get the first address for that wallet
+        let (Right seed) = fromMnemonic @'[15] mnemonics
+        let (Right pwd) = fromText "cardano-wallet"
+        let addr0 = firstAddress (seed, pwd)
```
I suppose It would be actually quite useful to implement the `listAddress` endpoint and use it for this...

KtorZ replied:
Hmmm.. Yes, and no. The wallets above aren't in the API _yet_. That's the whole point, we simply create a transaction to the corresponding addresses, and, if restored, the wallets will already have some funds :)

KtorZ replied:
(would be clearer with the next incoming PR, but generally speaking, I do agree about implementing the `listAddress` endpoint. I think I'll tackle that as part of the debt).

KtorZ committed:
Add 'Ord' instance for 'WalletState'

KtorZ committed:
refactor temporary transaction scenario & provide a 'fixtureWallet' utility to generate wallet with funds

KtorZ committed:
review WalletLayer interface implementation indentation

Moved all function implementations under a 'where' clause for better readability. This also allows for doing some extra
bootstraping prior to creating the wallet layer

KtorZ committed:
cleanup restoration workers when wallet is deleted

piotr-iohk reviewed (approved):
lgtm :+1:
